hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java

```diff
@@ -325,6 +324,42 @@ private void buildApiNodeEdges(StructuredTraceGraph graph) {
           }
         }
       }
+      // Sometimes an exit span might be missing for services like Istio, Kong.
+      // Only Entry spans will be populated for these services,
+      // an edge must be created between these services as well.
+      // 1. get entry boundary event from an api node.
+      // 2. find all the children of entry boundary event, which will be entry boundary nodes of
+      //  different api nodes
+      // 3. Check both parent span and child belong to different service as well.
+      // 4. connect the entry boundary of different api nodes with an edge.
+      Optional<Event> entryBoundaryEvent = apiNode.getEntryApiBoundaryEvent();
+      if (entryBoundaryEvent.isPresent()) {
+        List<Event> children = graph.getChildrenEvents(entryBoundaryEvent.get());
+        if (children != null) {
+          for (Event child : children) {
+            // if the child of an entry boundary event is an entry api boundary type,
+            // which can happen if exit span missing and both belongs to different services.
+            if (EnrichedSpanUtils.isEntryApiBoundary(child)
```

Comment:
if the exit span is missing how will these two span have parent and child relationship?

Reply:
Ignore this, I was assuming that an actual span is missing here which is not the case